### PR TITLE
fix(web): node container log stream sharing

### PR DESCRIPTION
**Pull request by m8vago**
- Remove NodeGlobalContainerController
- Add `_` as global container namespace.
- Refactor agent calls for log streaming
- Add upgradeable log stream watcher
- Automatically reattach to the log stream, when an agent reconnects

### CODE REVIEW

File: golang/internal/grpc/grpc.go [hunk 1]
```diff
@@ -461,36 +462,36 @@ func executeVersionDeployRequest(
 	}
 }
 
-func streamContainerStatus(
+func streamContainerState(
```
Comment (nandor-magyar) [inline golang/internal/grpc/grpc.go:465]: Both Kubernetes and Docker uses Status to describe workload progress, why the change?
In general terms I would say, state is something we will alter, status is something external to ourselves that we are aware of. I'm open for discussion.

File: web/crux/.env.example [hunk 2]
```diff
@@ -60,8 +60,8 @@ DISABLE_RECAPTCHA=true
 # Required only when ReCAPTCHA is enabled
 RECAPTCHA_SECRET_KEY=<recaptcha_secret_key>
 
-# Determines the quantity of lines being returned by a Docker container, while streaming its logs
-DEFAULT_CONTAINER_LOG_TAIL=100
+# Determines the maximum quantity of lines returned by a container
```
Comment (nandor-magyar) [inline web/crux/.env.example:63]: The phrasing is incorrect, the logs are not returned "by" the container. It could either be "from" or "of".

**Review by nandor-magyar — CHANGES_REQUESTED** [review]
I feel like I have got a fresh pair of eyes :eyes: 
I've checked the Go part, that looks good. :+1: My comments are mostly wording related.
I've skimmed the TS part too, but that would better be checked by some else.

File: golang/internal/grpc/grpc.go [hunk 3]
```diff
@@ -53,7 +53,7 @@ type ContainerLogContext struct {
 	Echo   bool
 }
 
-type ContainerWatchContext struct {
+type ContainerStateContext struct {
```
[the remark this answers is not in the document]
Comment (m8vago) [reply golang/internal/grpc/grpc.go:56]: Any suggestion?

File: golang/internal/grpc/grpc.go (hunk 1 above)
Comment (m8vago) [reply golang/internal/grpc/grpc.go:465]: Docker definitely uses state:
```// ContainerState stores container's running state
// it's part of ContainerJSONBase and will return by "inspect" command
type ContainerState struct {
	Status     string // String representation of the container state. Can be one of "created", "running", "paused", "restarting", "removing", "exited", or "dead"
	Running    bool
	Paused     bool
	Restarting bool
	OOMKilled  bool
	Dead       bool
	Pid        int
	ExitCode   int
	Error      string
	StartedAt  string
	FinishedAt string
	Health     *Health `json:",omitempty"`
}```

Even though we convert that to a more simple structure,
```
export const CONTAINER_STATE_VALUES = ['running', 'waiting', 'exited', 'removed'] as const
export type ContainerState = (typeof CONTAINER_STATE_VALUES)[number]
```
we use State in the whole stack, so basically this one is the oddly named.

**Review by robot9706 — APPROVED** [review]

File: golang/internal/grpc/grpc.go (hunk 1 above)
Comment (nandor-magyar) [reply golang/internal/grpc/grpc.go:465]: Docker does it well too. They actually manage those states, they mutate those values. If you look deeper, in that State object, you can find 'statuses', because those are external to them, since they describe external processes. So they also follow this logic, as Kubernetes. We used it vice-versa, @robot9706 started correcting them not so long ago.

File: golang/internal/grpc/grpc.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (nandor-magyar) [reply golang/internal/grpc/grpc.go:56]: It contains channels only, so it could have the suffix of Stream, Chan or Channels.

File: golang/internal/grpc/grpc.go (hunk 1 above)
Comment (m8vago) [reply golang/internal/grpc/grpc.go:465]: So what is your proposal?

Comment (m8vago) [reply golang/internal/grpc/grpc.go:465]: Allrighty.
